docs/IR.md

```diff
@@ -1,127 +1,353 @@
+### Tensor Element Types
 
+|Group|Types|Description| 
+|---|---|---|
+Floating Point Types|float16, float32, float64|Values adhering to the IEEE 754-2008 standard representation of floating-point data.
+Signed Integer Types|int8, int16, int32, int64|Signed integers are supported for 8-64 bit widths.
+Unsigned Integer Types|uint8, uint16|Unsigned integers of 8 or 16 bits are supported.
+Complex Types|complex64, complex128|A complex number with either 32- or 64-bit real and imaginary parts.
+Other|string|Strings represent textual data. All strings are encoded using UTF-8.
+Other|bool|Boolean value represent data with only two values, typically true and false.
 
-### Standard data types
+### Input / Output Data Types
 
-The following data types are supported by ONNX. Additional data types can be supported by frameworks.
+The following types are used to define the types of graph and node inputs and outputs.
 
-|Group|Name|Description|
-|-----|----|-----------|
-|Floating Point Types|__float16, float (32 bit), double (64 bit)__|Values adhering to the IEEE 754-2008 standard representation of floating-point data.|
-|Signed Integer Types|__int8, int16,int32,int64__|Signed integers are supported for 8-64 bit widths.|
-|Unsigned Integer Types|__uint8,uint16__| Unsigned integers of 8 or 16 bits are supported.|
-|Complex Types|__complex64,complex128__|A complex number with either 32- or 64-bit real and imaginary parts.|
-|Other|__string__|Strings represent textual data. All strings are encoded using UTF-8.|
-|Other|__bool__|Boolean value represent data with only two values, typically _true_ and _false_.|
-|Other|__handle__|Handles are opaque types holding a 64-bit integer.|
-|Collections|__sparse and dense tensor__|Tensors are a generalization of vectors and matrices; whereas vectors have one dimension, and matrices two, tensors can have any number of dimensions, including zero. A zero-dimensional tensor is equivalent to a scalar.|
+|Variant | Type | Description | 
+|---|---|---|
+ONNX|dense tensors|Tensors are a generalization of vectors and matrices; whereas vectors have one dimension, and matrices two, tensors can have any number of dimensions, including zero. A zero-dimensional tensor is logically equivalent to a scalar value.
+ONNX-ML|sequence|Sequences represent dense, ordered, collections of elements that are of homogeneous types.
+ONNX-ML|map|Maps represent associative tables, defined by a key type and a value type.
 
-Extensibility
--------------
+ONNX currently does not define a sparse tensor type.
 
-ONNX is expected to evolve over time and provides features that enable users to experiment and implement additions before they are added to the specifications.
+#### Tensor shapes
 
-### Metadata
+In addition to element type, tensors have shape. A tensor shape is a list of records that define whether the tensor is a vector, a matrix, or a higher-dimensional value. For example, a 100x100 matrix has the shape [100,100].
```

Comment:
How about "A tensor shape is a list of dimensions"?

Reply:
Yes, I agree that it's better.